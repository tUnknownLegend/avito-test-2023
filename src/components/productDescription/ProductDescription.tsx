import {Descriptions, DescriptionsProps} from 'antd';

// const items: DescriptionsProps['items'] = [
//     {
//         key: '1',
//         label: 'UserName',
//         children: 'Zhou Maomao',
//     },
//     {
//         key: '2',
//         label: 'Telephone',
//         children: '1810000000',
//     },
//     {
//         key: '3',
//         label: 'Live',
//         children: 'Hangzhou, Zhejiang',
//     },
//     {
//         key: '4',
//         label: 'Remark',
//         children: 'empty',
//     },
//     {
//         key: '5',
//         label: 'Address',
//         children: 'No. 18, Wantang Road, Xihu District, Hangzhou, Zhejiang, China',
//     },
// ];

function ProductDescription({title, data} : {title: string, data: object}) {
    const items: DescriptionsProps['items'] = Object.keys(data).map((entry, index: number) => {
        return {
            key: index,
            label: entry.charAt(0).toUpperCase() + entry.slice(1).split('_').join(' '),
            children: data[entry] ?? 'Not defined',
        };
    });
    return (
        <Descriptions title={title} items={items} />
    );
}

export default ProductDescription;
