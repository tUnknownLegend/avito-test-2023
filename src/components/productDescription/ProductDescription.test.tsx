import {render} from '@testing-library/react';
import ProductDescription from './ProductDescription.tsx';

afterEach(jest.clearAllMocks);

const firstTestItem = {
    className: 'product-system-requirements',
    title: 'System requirements',
    data: {
        os: 'Windows XP ',
        processor: '?',
        memory: '2 GB RAM ',
        graphics: 'Integrated graphics is fine',
        storage: '200 MB available space ',
    },
};

describe('Product Description test', () => {
    beforeAll(() => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation((query) => ({
                matches: false,
                media: query,
                onchange: null,
                addListener: jest.fn(), // Deprecated
                removeListener: jest.fn(), // Deprecated
                addEventListener: jest.fn(),
                removeEventListener: jest.fn(),
                dispatchEvent: jest.fn(),
            })),
        });
    });

    it('should render correctly PC version', () => {
        const rendered = render(
            <ProductDescription
                title={firstTestItem.title}
                className={firstTestItem.className}
                data={firstTestItem.data}
            />,
        );

        expect(rendered.asFragment()).toMatchSnapshot();
    });
    it('should render correctly Browser version', () => {
        const rendered = render(
            <ProductDescription
                title={firstTestItem.title}
                className={firstTestItem.className}
                data={{}}
            />,
        );

        expect(rendered.asFragment()).toMatchSnapshot();
    });
});
