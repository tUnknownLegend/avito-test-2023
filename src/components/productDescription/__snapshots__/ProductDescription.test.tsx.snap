// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Description test should render correctly Browser version 1`] = `
<DocumentFragment>
  <div
    class="ant-descriptions product-system-requirements css-dev-only-do-not-override-17a39f8"
  >
    <div
      class="ant-descriptions-header"
    >
      <div
        class="ant-descriptions-title"
      >
        System requirements
      </div>
    </div>
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody />
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Product Description test should render correctly PC version 1`] = `
<DocumentFragment>
  <div
    class="ant-descriptions product-system-requirements css-dev-only-do-not-override-17a39f8"
  >
    <div
      class="ant-descriptions-header"
    >
      <div
        class="ant-descriptions-title"
      >
        System requirements
      </div>
    </div>
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <td
              class="ant-descriptions-item"
              colspan="1"
            >
              <div
                class="ant-descriptions-item-container"
              >
                <span
                  class="ant-descriptions-item-label"
                >
                  Os
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  Windows XP 
                </span>
              </div>
            </td>
            <td
              class="ant-descriptions-item"
              colspan="1"
            >
              <div
                class="ant-descriptions-item-container"
              >
                <span
                  class="ant-descriptions-item-label"
                >
                  Processor
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  ?
                </span>
              </div>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <td
              class="ant-descriptions-item"
              colspan="1"
            >
              <div
                class="ant-descriptions-item-container"
              >
                <span
                  class="ant-descriptions-item-label"
                >
                  Memory
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  2 GB RAM 
                </span>
              </div>
            </td>
            <td
              class="ant-descriptions-item"
              colspan="1"
            >
              <div
                class="ant-descriptions-item-container"
              >
                <span
                  class="ant-descriptions-item-label"
                >
                  Graphics
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  Integrated graphics is fine
                </span>
              </div>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <td
              class="ant-descriptions-item"
              colspan="2"
            >
              <div
                class="ant-descriptions-item-container"
              >
                <span
                  class="ant-descriptions-item-label"
                >
                  Storage
                </span>
                <span
                  class="ant-descriptions-item-content"
                >
                  200 MB available space 
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
